pkg/virt-api/rest/subresource.go

```diff
@@ -420,37 +510,40 @@ func (app *SubresourceAPIApp) StopVMRequestHandler(request *restful.Request, res
 	response.WriteHeader(http.StatusAccepted)
 }
 
-func (app *SubresourceAPIApp) findPod(namespace string, vmi *v1.VirtualMachineInstance) (string, error) {
-	fieldSelector := fields.ParseSelectorOrDie("status.phase==" + string(k8sv1.PodRunning))
-	labelSelector, err := labels.Parse(fmt.Sprintf(v1.AppLabel + "=virt-launcher," + v1.CreatedByLabel + "=" + string(vmi.UID)))
-	if err != nil {
-		return "", err
+func (app *SubresourceAPIApp) findVirtHandlerPodForVMI(vmi *v1.VirtualMachineInstance) (*k8sv1.Pod, error) {
+	if !vmi.IsRunning() {
```

Comment:
Can you take the node name from the status of the vmi and do `kubecli.NewVirtHandlerClient(app.VirtCli).ForNode(vmi.Status.NodeName).ConsoleURI(vmi)`? That should pretty much do what this method does. You will have to adjust the connection URI in `ConsoleURI` since it is a left-over which we forgot to remove, but apart from that it should give you waht you need. You can add a `VNCURI(vmi)` method there.

Reply:
Done